src/AzureOpenAIProxy.ApiApp/Builders/OpenAISettingsBuilder.cs

```diff
@@ -0,0 +1,31 @@
+﻿using AzureOpenAIProxy.ApiApp.Configurations;
+
+namespace AzureOpenAIProxy.ApiApp.Builders;
+
+/// <summary>
+/// This provides interface to the <see cref="OpenAISettingsBuilder"/> class.
+/// </summary>
+public interface IOpenAISettingsBuilder
+{
+    /// <summary>
+    /// Builds the <see cref="OpenAISettings"/> instance.
+    /// </summary>
+    /// <returns>Returns the <see cref="OpenAISettings"/> instance.</returns>
+    OpenAISettings Build();
+}
+
+/// <summary>
+/// This represents the builder entity for <see cref="OpenAISettings"/> class.
+/// </summary>
+public class OpenAISettingsBuilder : IOpenAISettingsBuilder
+{
+    private OpenAISettings? _settings;
```

Comment:
There's no way to set this field value. Maybe changing it to the list of `OpenAIInstanceSettings` and inject it from outside?